lib/omnibus/publishers/s3_publisher.rb

```diff
@@ -63,19 +63,20 @@ def s3_configuration
     # The unique upload key for this package. The additional "stuff" is
     # postfixed to the end of the path.
     #
+    # @example
+    #   'el/6/x86_64/chef-11.6.0-1.el6.x86_64.rpm/chef-11.6.0-1.el6.x86_64.rpm'
+    #
     # @param [Package] package
     #   the package this key is for
     # @param [Array<String>] stuff
-    #   the additional things to prepend
+    #   the additional things to append
     #
     # @return [String]
     #
     def key_for(package, *stuff)
+      pattern = Config.publish_dir_pattern || "%{platform}/%{platform_version}/%{arch}/%{basename}"
       File.join(
-        package.metadata[:platform],
-        package.metadata[:platform_version],
-        package.metadata[:arch],
-        package.name,
```

Comment:
[Q] I'm not sure about this -- for this to work the same as before, `package.name == package.metadata[:basename]` must hold; is that the case? (It's also possible that I'm just confused.)